pkg/svcat/service-catalog/broker.go

```diff
@@ -109,8 +112,49 @@ func (sdk *SDK) RetrieveBroker(name string) (*v1beta1.ClusterServiceBroker, erro
 	return broker, nil
 }
 
-// RetrieveNamespacedBroker gets a broker by its name & namespace.
-func (sdk *SDK) RetrieveNamespacedBroker(namespace string, name string) (*v1beta1.ServiceBroker, error) {
+// RetrieveBrokerByID gets a broker by its k8s name
+func (sdk *SDK) RetrieveBrokerByID(kubeName string, opts ScopeOptions) (Broker, error) {
+	var csb *v1beta1.ClusterServiceBroker
+	var sb *v1beta1.ServiceBroker
+	var err error
+	if opts.Scope.Matches(ClusterScope) {
+		csb, err = sdk.ServiceCatalog().ClusterServiceBrokers().Get(kubeName, v1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			csb = nil
+		}
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("unable to search clusterservicebrokers by k8s name (%s)", err)
+		}
+	}
+
+	if opts.Scope.Matches(NamespaceScope) {
+		sb, err = sdk.ServiceCatalog().ServiceBrokers(opts.Namespace).Get(kubeName, v1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				sb = nil
+			}
+			if err != nil && !apierrors.IsNotFound(err) {
+				return nil, fmt.Errorf("unable to search servicebrokers k8s by name (%s)", err)
+			}
+		}
+	}
+
+	switch {
+	case csb != nil && sb != nil:
+		return nil, fmt.Errorf(MultipleBrokersFoundError+" for '%s'", kubeName)
+	case csb == nil && sb == nil:
+		return nil, fmt.Errorf("no matching broker found for k8s name '%s'", kubeName)
+	case csb != nil && sb == nil:
+		return csb, nil
+	case csb == nil && sb != nil:
+		return sb, nil
```

Comment:
why not return earlier? I don't understand the construction here that would allow this.